app/src/main/java/org/mozilla/firefox/firefoxlitedownloader/MainActivity.java

```diff
@@ -1,50 +1,476 @@
+    }
+
+    @Override
+    public void onTimeSet(TimePicker view, int scheduledhours, int minute) {
+        Calendar rightNow = Calendar.getInstance();
+        int currentHourIn24Format = rightNow.get(Calendar.HOUR_OF_DAY);
+        if (currentHourIn24Format==0){
+            currentHourIn24Format=24;
+        }
+        int diffhours=0;
+        int currentminutes = rightNow.get(Calendar.MINUTE);
+        if(scheduledhours>currentHourIn24Format){
+            diffhours=scheduledhours-currentHourIn24Format;
+        }
+        else if (scheduledhours==currentHourIn24Format){
+            diffhours=0;
+        }
+        else{
+            diffhours=(24-currentHourIn24Format)+scheduledhours;
+        }
+        int diffminutes=Math.abs(minute-currentminutes);
+        int totaldiffminutes=diffhours*60+diffminutes;
+        Toast.makeText(getApplicationContext(),totaldiffminutes+"",Toast.LENGTH_SHORT).show();
+        downloadIdOne = MozzDownloader.download(URL1, dirPath, "mandys.pmd")
+                .build().setOnProgressListener(new OnProgressListener() {
+                    @Override
+                    public void onProgress(Progress progress) {
+                        long progressPercent = progress.currentBytes * 100 / progress.totalBytes;
+                        progressBarOne.setProgress( (int)progressPercent);
+                        textViewProgressOne.setText(Utils.getProgressDisplayLine(progress.currentBytes, progress.totalBytes));
+                        progressBarOne.setIndeterminate(false);
+                    }
+                }).setOnStartOrResumeListener(new OnStartOrResumeListener() {
+                    @Override
+                    public void onStartOrResume() {
+                        progressBarOne.setIndeterminate(false);
+                        buttonOne.setEnabled(true);
+                        buttonOne.setText(R.string.pause);
+                        buttonCancelOne.setEnabled(true);
+                    }
+                })
+                .setOnPauseListener(new OnPauseListener() {
+                    @Override
+                    public void onPause() {
+                        buttonOne.setText(R.string.resume);
+                    }
+                })
+                .setOnCancelListener(new OnCancelListener() {
                     @Override
-                    public void onChanged(@Nullable Integer integer) {
-                        progressBarDownload.setProgress(integer);
+                    public void onCancel() {
+                        buttonOne.setText(R.string.start);
+                        buttonCancelOne.setEnabled(false);
+                        progressBarOne.setProgress(0);
+                        textViewProgressOne.setText("");
+                        downloadIdOne = 0;
+                        progressBarOne.setIndeterminate(false);
                     }
-                });
+                }).schedule(getApplicationContext(),mainActivity,totaldiffminutes);
```

Comment:
passing mainActivity to other library is dangerous if you hold an activity reference there.